internal/infra/table/sql_extractor.go

```diff
@@ -128,3 +146,87 @@ func (e *SQLExtractor) Count(tableName string) (int, *table.Error) {
 
 	return count, nil
 }
+
+func (e *SQLExtractor) ColumnInfo(db *sql.DB, tableName string) ([]table.Column, error) {
+	// Execute query to fetch column information
+	query := e.dialect.Select(tableName)
+	rows, err := db.Query(query)
+	if err != nil {
+		return []table.Column{}, err
+	}
+	defer rows.Close()
+
+	// Retrieve column information
+	columnTypes, err := rows.ColumnTypes()
+	if err != nil {
+		return []table.Column{}, err
+	}
+
+	columns := []table.Column{}
+	columnsNoType := []string{}
+	// Iterate over column information
+	for _, ct := range columnTypes {
+		columnName := ct.Name()
+		dataType := ct.DatabaseTypeName()
+
+		// columnLength, _ := ct.Length()
+		// columnPrecision, columnSize, _ := ct.DecimalSize()
+		// if columnLength > 0 {
+		// 	fmt.Printf(", Length: %d", columnLength)
+		// } else if columnSize > 0 {
+		// 	fmt.Printf(", Size: %d", columnSize)
+		// 	fmt.Printf(", Precision: %d", columnPrecision)
+		// }
+
+		// if data type is unusual or data not correct
+		if len(dataType) == 0 {
+			columnsNoType = append(columnsNoType, columnName)
+		}
+
+		exportType, needExport := checkType(dataType)
+		columnInfo := table.Column{
+			Name: columnName,
+		}
+
+		if needExport {
+			columnInfo.Export = exportType
+		}
+
+		columns = append(columns, columnInfo)
+	}
+
+	// Notify user unusual column
+	if len(columnsNoType) > 0 {
+		msg := fmt.Sprintf("Table %s contains some columns with unusual characteristics: %v. It may be necessary to manually specify the export type if the data does not display correctly.", tableName, columnsNoType)
+		log.Warn().Msg(msg)
```

Comment:
These line can be done in a single call to the log package. See Msgf()` method, example:

```

    log.Fatal().
        Err(err).
        Str("service", service).
        Msgf("Cannot start %s", service)
```